timemachine/cpp/src/potential.cu

```diff
@@ -11,28 +11,42 @@ namespace timemachine {
 
 const int Potential::D = 3;
 
-void Potential::execute_coords_batch_host(
-    const int K,
-    const int N,
-    const int P,
-    const double *h_x,           // [K, N,3]
-    const double *h_p,           // [P,]
-    const double *h_box,         // [K, 3, 3]
+// COORDS_BATCH flag of true indicates to batch over coords, false indicates to batch over parameters
+template <bool COORDS_BATCH>
```

Comment:
Enum seems preferable, to handle case of batching over lambda.

Another batch style we may want to distinguish here:
* `[NonbondedInteractionGroup(x_lig, x_env) for x_lig in lig_coords_batch]` (with fixed `x_env`) vs.
* `[NonbondedInteractionGroup(x[lig], x[env]) for x in full_coords_batch]`

Reply:
Regarding the `NonbondedInteractionGroup` methods, do you mean that `x_lig` is different coords idxs or just where the ligand atoms are all that change? In the latter case this implementation should perform the same. The NBlist will be rebuilt as long as a single atom has moved enough, whether that is a host or ligand atom.

Reply:
> Regarding the NonbondedInteractionGroup methods, do you mean that x_lig is different coords idxs or just where the ligand atoms are all that change?

Second case

>In the latter case this implementation should perform the same. The NBlist will be rebuilt as long as a single atom has moved enough, whether that is a host or ligand atom.

Even if nblist rebuild condition is identical, I guessed one might still be cheaper than the other since there's much less data to transfer...

Point isn't to suggest applying optimizations for this case, but to confirm there are likely to be >2 relevant batch modes -- enum seems preferable to bool.